### PR TITLE
CI: update cilium-cli to v0.12.10, force deploy connectivity test pods on GKE

### PR DESCRIPTION
**Pull request by tklauser**
See individual commit messages.

For #22368

### CODE REVIEW

**Comment by tklauser** [comment]
External workloads workflow is failing with

> Error: Unable to create external workload install script: No Major.Minor.Patch elements found

https://github.com/cilium/cilium/actions/runs/3582806205/jobs/6027497979

I assume this is due to https://github.com/cilium/cilium-cli/pull/1238 /cc @brb

**Comment by tklauser** [comment]
GKE workflow passed though: https://github.com/cilium/cilium/actions/runs/3582806210/jobs/6027502665

Re-running to verify that https://github.com/cilium/cilium/issues/22368 is indeed fixed.

File: .github/workflows/conformance-gke-v1.10.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ on:
   # - Don't forget to remove the `DO NOT MERGE` commit once satisfied. The run
   #   will disappear from the PR checks: please provide a direct link to the
   #   successful workflow run (can be found from Actions tab) in a comment.
-  # 
+  #
```
Comment (nbusseneau) [inline .github/workflows/conformance-gke-v1.10.yaml:19]: 🕵️‍♀️

**Review by nbusseneau — APPROVED** [review]

**Review by pchaigno — APPROVED** [review]
LGTM assuming the last commit is removed of course.

IMO, it's fine to sacrifice External Workloads temporarily to reenable GKE. The latter is more important/more likely to catch regressions.

**Comment by tklauser** [comment]
Removed the test commit and marking as ready to merge given that reviews are in and all workflows except external workloads passed (namely GKE) passed. Let's disable external workloads workflow for now and re-enable once we switch to a cilium-cli version with a fix.

**Pull request merged**